### ISSUE TITLE
`atlas kubernetes generate ...` manifests issues in Deployment

### ISSUE THREAD

**Issue by zelig81** [mongodb/mongodb-atlas-kubernetes#855]
**What did you do to encounter the bug?**
- `atlas kubernetes config generate --targetNamespace atlas --projectId <id> --clusterName <cluster-name>`
- project is called in uppercase (POC_PROJECT)
- version of DB is 4.4 (created from browser)
- read the Operator documentation :) 

**What did you expect?**
- get in k8s manifest project id in uppercase
- get in k8s manifest version of DB
- find via documentation where exactly to put a version of DB (4.4 in my case)

**What happened instead?**
A clear and concise description of what happened instead
- project id reference is in lowercase (operator complains that the project is not found)
- no version of DB (see in the attached manifest)
- in documentation I found a reference to .spec.deploymentSpec.mongoDBMajorVersion, while atlas cli generated only .spec.advancedDeploymentSpec without version... So I'm not sure which one to use (or documentation is not updated, or cli did not catch version)

**Operator Information**
 - Operator Version: v1.5.1

**Kubernetes Cluster Information**
 - Distribution: GKE 1.25
 - Version: 1.25
 - Image Registry location: default

**Additional context**

<details><summary>DB manifest</summary>

```yaml
apiVersion: atlas.mongodb.com/v1
kind: AtlasDeployment
metadata:
  creationTimestamp: null
  name: cluster0
  namespace: atlas
spec:
  advancedDeploymentSpec:
    backupEnabled: true
    biConnector:
      enabled: false
      readPreference: secondary
    clusterType: REPLICASET
    encryptionAtRestProvider: NONE
    name: cluster0
    paused: false
    pitEnabled: true
    replicationSpecs:
    - numShards: 1
      regionConfigs:
      - analyticsSpecs:
          instanceSize: M10
          nodeCount: 0
        autoScaling:
          compute:
            enabled: true
            maxInstanceSize: M30
            minInstanceSize: M10
            scaleDownEnabled: true
          diskGB:
            enabled: true
        electableSpecs:
          instanceSize: M10
          nodeCount: 3
        priority: 7
        providerName: GCP
        readOnlySpecs:
          instanceSize: M10
          nodeCount: 0
        regionName: EASTERN_US
      zoneName: Zone 1
    rootCertType: ISRGROOTX1
    versionReleaseSystem: LTS
  backupRef:
    name: ""
    namespace: ""
  processArgs:
    javascriptEnabled: true
    minimumEnabledTlsProtocol: TLS1_2
    noTableScan: false
  projectRef:
    name: poc_project
    namespace: atlas
status:
  conditions: []
```

</details>

<details><summary>operator error</summary>

`{"level":"INFO","time":"2023-01-25T14:32:58.877Z","msg":"Status update","atlasdeployment":"atlas/cluster0","lastCondition":{"type":"DeploymentReady","status":"False","lastTransitionTime":null,"reason":"InternalError","message":"AtlasProject.atlas.mongodb.com \"poc_project\" not found"}}`

</details>

**Comment by igor-karpukhin** [comment]
Hi @zelig81. Resource names for k8s objects are always in lower case as well as they don't have to contain certain characters. See RFC1123. Could you please also post the `AtlasProject` resource that was generated?

**Comment by zelig81** [comment]
Sorry, you are right:

<details><summary>AtlasProject</summary>

```yaml
apiVersion: atlas.mongodb.com/v1
kind: AtlasProject
metadata:
  creationTimestamp: null
  name: poc_project
  namespace: atlas
spec:
  alertConfigurations:
  - enabled: true
    eventTypeName: REPLICATION_OPLOG_WINDOW_RUNNING_OUT
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold:
      operator: LESS_THAN
      threshold: "1.000000"
      units: HOURS
  - enabled: true
    eventTypeName: NO_PRIMARY
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 15
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: CLUSTER_MONGOS_IS_MISSING
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 15
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_METRIC_THRESHOLD
    metricThreshold:
      metricName: CONNECTIONS_PERCENT
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "80.000000"
      units: RAW
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_METRIC_THRESHOLD
    metricThreshold:
      metricName: DISK_PARTITION_SPACE_USED_DATA
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "90.000000"
      units: RAW
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_METRIC_THRESHOLD
    metricThreshold:
      metricName: QUERY_TARGETING_SCANNED_OBJECTS_PER_RETURNED
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "1000.000000"
      units: RAW
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: CREDIT_CARD_ABOUT_TO_EXPIRE
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 1440
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_METRIC_THRESHOLD
    metricThreshold:
      metricName: NORMALIZED_SYSTEM_CPU_USER
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "95.000000"
      units: RAW
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: HOST_HAS_INDEX_SUGGESTIONS
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 10
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: HOST_MONGOT_CRASHING_OOM
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_SERVERLESS_METRIC_THRESHOLD
    metricThreshold:
      metricName: SERVERLESS_CONNECTIONS_PERCENT
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "80.000000"
      units: RAW
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_SERVERLESS_METRIC_THRESHOLD
    metricThreshold:
      metricName: SERVERLESS_DATA_SIZE_TOTAL
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "0.750000"
      units: TERABYTES
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: HOST_NOT_ENOUGH_DISK_SPACE
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: JOINED_GROUP
    metricThreshold:
      threshold: ""
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_SERVERLESS_METRIC_THRESHOLD
    metricThreshold:
      metricName: SERVERLESS_TOTAL_READ_UNITS
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "0.250000"
      units: MILLION_RPU
    notifications:
    - delayMin: 30
      emailEnabled: true
      intervalMin: 720
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_SERVERLESS_METRIC_THRESHOLD
    metricThreshold:
      metricName: SERVERLESS_TOTAL_READ_UNITS
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "1.000000"
      units: MILLION_RPU
    notifications:
    - delayMin: 5
      emailEnabled: true
      intervalMin: 120
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  - enabled: true
    eventTypeName: OUTSIDE_METRIC_THRESHOLD
    metricThreshold:
      metricName: SEARCH_MAX_NUMBER_OF_LUCENE_DOCS
      mode: AVERAGE
      operator: GREATER_THAN
      threshold: "1.000000"
      units: BILLION
    notifications:
    - delayMin: 0
      emailEnabled: true
      intervalMin: 60
      roles:
      - GROUP_OWNER
      smsEnabled: false
      typeName: GROUP
    threshold: {}
  auditing:
    auditAuthorizationSuccess: false
    enabled: false
  connectionSecretRef:
    name: ""
  encryptionAtRest:
    awsKms:
      enabled: false
      valid: false
    azureKeyVault:
      enabled: false
    googleCloudKms:
      enabled: false
  maintenanceWindow: {}
  name: POC_PROJECT
  projectIpAccessList:
  - cidrBlock: 0.0.0.0/0
  settings:
    isCollectDatabaseSpecificsStatisticsEnabled: true
    isDataExplorerEnabled: true
    isPerformanceAdvisorEnabled: true
    isRealtimePerformancePanelEnabled: true
    isSchemaAdvisorEnabled: true
status:
  conditions: []
```

</details>

**Comment by zelig81** [comment]
My fault - I did not check the result of `kubectl apply -k ...` and there I had the following error:
`The AtlasProject "poc_project" is invalid: metadata.name: Invalid value: "poc_project": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')`

I'll do the changes manually - this issue should be opened in mongodb-atlas-cli :(

**Comment by zelig81** [comment]
I got a next error (I'll add it to cli repo as well):

`{"level":"INFO","time":"2023-01-26T08:26:59.370Z","msg":"Status update","atlasdeployment":"atlas/cluster0","lastCondition":{"type":"DeploymentReady","status":"False","lastTransitionTime":null,"reason":"InternalError","message":"groupId is invalid because must be set"}}`

**Comment by igor-karpukhin** [comment]
Hi @zelig81. This is fixed by this PR: https://github.com/mongodb/mongodb-atlas-cli/pull/1631. We'll let you know when it is released

**Comment by zelig81** [comment]
After a change in project name in k8s manifests: `poc_project` -> `poc-project` while Atlas project name remains the same POC_PROJECT, I see the following error about authorization: `{"level":"INFO","time":"2023-01-26T09:05:58.153Z","msg":"Error","atlasproject":"atlas/qmp-dev","err":"GET https://cloud.mongodb.com/api/atlas/v1.0/groups/byName/POC_PROJECT: 401 (request \"\") You are not authorized for this resource."}`. The permissions of the api key are the same as before (I'll check it once more anyway)

in deployment:

```yaml
  status:
    conditions:
    - lastTransitionTime: "2023-01-25T14:22:17Z"
      status: "False"
      type: Ready
    - lastTransitionTime: "2023-01-25T14:22:17Z"
      status: "True"
      type: ValidationSucceeded
    - lastTransitionTime: "2023-01-25T14:22:17Z"
      message: groupId is invalid because must be set
      reason: InternalError
      status: "False"
      type: DeploymentReady
```

in project:

```yaml
status:
  conditions:
  - lastTransitionTime: "2023-01-26T08:26:38Z"
    status: "False"
    type: Ready
  - lastTransitionTime: "2023-01-26T08:26:38Z"
    status: "True"
    type: ValidationSucceeded
  - lastTransitionTime: "2023-01-26T08:26:38Z"
    message: 'GET https://cloud.mongodb.com/api/atlas/v1.0/groups/byName/POC_PROJECT:
      401 (request "") You are not authorized for this resource.'
    reason: ProjectNotCreatedInAtlas
    status: "False"
    type: ProjectReady
```

**Comment by zelig81** [comment]
> Hi @zelig81. This is fixed by this PR: [mongodb/mongodb-atlas-cli#1631](https://github.com/mongodb/mongodb-atlas-cli/pull/1631). We'll let you know when it is released

Thank you!!!!

**Comment by igor-karpukhin** [comment]
The fix is released with atlas-cli 1.5.0: https://github.com/mongodb/mongodb-atlas-cli/releases/tag/atlascli%2Fv1.5.0.

**Comment by igor-karpukhin** [comment]
Closing this one as fixed with AtlasCLI v1.5.0
